test/schemas/v0-0-1/tests.schema.json

```diff
@@ -0,0 +1,334 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://raw.githubusercontent.com/unicode-org/message-format-wg/main/test/schemas/v0-0-1/tests.schema.json",
+  "title": "MessageFormat 2 data-driven tests",
+  "description": "This is the main schema for MessageFormat 2 test source data.",
+  "type": "object",
+  "additionalProperties": false,
+  "required": [
+    "tests"
+  ],
+  "properties": {
+    "scenario": {
+      "type": "string",
+      "description": "Identifier for the tests in the file"
+    },
+    "description": {
+      "type": "string",
+      "description": "Information about the test scenario."
+    },
+    "defaultTestProperties": {
+      "$ref": "#/$defs/defaultTestProperties"
+    },
+    "tests": {
+      "type": "array",
+      "items": {
+        "$ref": "#/$defs/test"
+      },
+      "minItems": 1
+    }
+  },
+  "$comment": "This allOf specifies required test properties that allow a default. A value will be required in 'defaultTestProperties' if one is not provided for every individual test.",
+  "allOf": [
+    {
+      "anyOf": [
+        {
+          "properties": {
+            "defaultTestProperties": {
+              "required": [
+                "testType"
+              ]
+            }
+          }
+        },
+        {
+          "properties": {
+            "tests": {
+              "type": "array",
+              "items": {
+                "required": [
+                  "testType"
+                ]
+              }
+            }
+          }
+        }
+      ]
+    },
+    {
+      "anyOf": [
+        {
+          "properties": {
+            "defaultTestProperties": {
+              "required": [
+                "locale"
+              ]
+            }
+          }
+        },
+        {
+          "properties": {
+            "tests": {
+              "type": "array",
+              "items": {
+                "required": [
+                  "locale"
+                ]
+              }
+            }
+          }
+        }
+      ]
+    },
+    {
+      "anyOf": [
+        {
+          "properties": {
+            "defaultTestProperties": {
+              "required": [
+                "src"
+              ]
+            }
+          }
+        },
+        {
+          "properties": {
+            "tests": {
+              "type": "array",
+              "items": {
+                "required": [
+                  "src"
```

Comment:
Having `"src"` be either a string or an array breaks some json parsers used by strict type languages.

For example in Java one can use parsers with annotations, or reflection, or other methods to automatically parse the JSON into a class.

Using GSon with this simple class "just works", without any extra code:
```java
class Unit {
    final String src;
    final List<String> srcs;
    final String locale;
    final Map<String, Object> params;
    final String exp;
    final String ignore;
    final List<Error> errors;
```

GSon will automatically map a JSON like this to that class:
```json
{
   "src": "Hello {$user}!",
   "locale": "en-US",
   "params": { "user", "John" },
   "exp": "Hello John!"
}
```

If the schema allows for a field to be an either / or it breaks these kinds of parsers.
All of a sudden you get an `Object` or `JsonObject` back and you must do type checking and explicit handling of the various types.
From all I know these parsers are available (and common) for Java, C++ and Rust, probably others.

Reply:
This is one sort of problem users will experience in working with MF2 messages. Messages have to be stored in file and resource formats, typically as a string. We should ensure this dog food is yummy. 🐶🥫😋

Reply:
It seems like there's more deliberation to be had about the 'src' vs 'srcs' list. Since we're not using the list version yet, I'll leave it out of the schema for now. We can introduce it as an enhancement when needed and have more in-depth discussion then. Thanks for flagging, Mihai.